### PR TITLE
Upgrade all Gravitee dependencies and use GW Testing SDK

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

NA

**Description**

- upgrade all gravitee dependencies
- replace current integration tests by ones based on GW Testing SDK

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.6.3-update-deps-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-xslt/1.6.3-update-deps-SNAPSHOT/gravitee-policy-xslt-1.6.3-update-deps-SNAPSHOT.zip)

### CODE REVIEW

**Comment by gaetanmaisse** [comment]
Following the discussion we had during https://github.com/gravitee-io/gravitee-policy-xslt/pull/33 @a-cordier 😄

File: src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyResponseIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        awaitTerminalEvent(obs);
+        obs
+            .assertComplete()
+            .assertValue(response -> {
+                assertThat(response.statusCode()).isEqualTo(200);
+                String expectedResponse = loadResource("/io/gravitee/policy/xslt/valid-output01.xml");
+                assertThat(response.bodyAsString()).isEqualTo(expectedResponse);
+
+                MultiMap headers = response.headers();
+                assertThat(headers.contains(io.gravitee.common.http.HttpHeaders.CONTENT_TYPE)).isTrue();
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyResponseIntegrationTest.java:57]: Can you prefer HttpHeaderNames from gateway-api ? (for the other lines also :) )

**Review by ytvnr — APPROVED** [review]
just a remark

File: src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyRequestIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("OnRequestContent")
+@GatewayTest
+class XSLTTransformationPolicyRequestIntegrationTest extends AbstractXSLTTransformationPolicyIntegrationTest {
+
+    @Test
+    @DisplayName("Should transform and add header")
+    @DeployApi("/io/gravitee/policy/xslt/api-xslt-request-valid-stylesheet01.json")
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyRequestIntegrationTest.java:39]: In other tests with SDK, we use to place the apis to deploy in `test/resources/apis`

File: src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyRequestIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    void shouldTransformAndAddHeadersOnRequestContent(WebClient client) throws IOException {
+        wiremock.stubFor(post("/endpoint").willReturn(ok()));
+
+        String file01 = loadResource("/io/gravitee/policy/xslt/file01.xml");
+        final TestObserver<HttpResponse<Buffer>> obs = client.post("/test").rxSendBuffer(Buffer.buffer(file01)).test();
+
+        awaitTerminalEvent(obs);
+        obs
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyRequestIntegrationTest.java:47]: now `awaitTerminalEvent` returns the obs, so you are able to chain the calls directly :)

File: src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyRequestIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        );
+    }
+
+    @Test
+    @DisplayName("Should transform with parameters")
+    @DeployApi("/io/gravitee/policy/xslt/api-xslt-request-valid-stylesheet02.json")
+    void shouldTransformWithParametersOnRequestContent(WebClient client) throws IOException {
+        wiremock.stubFor(post("/endpoint").willReturn(ok()));
+
+        String file01 = loadResource("/io/gravitee/policy/xslt/file01.xml");
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyRequestIntegrationTest.java:71]: i would have prefer another name, like `xmlFile` maybe? (it's just a detail)

Comment (gaetanmaisse) [reply src/test/java/io/gravitee/policy/xslt/XSLTTransformationPolicyRequestIntegrationTest.java:71]: I followed the naming used in the unit tests, and as there is already the `xml` as an extension I'm not sure it's a super-valuable renaming 🤷🏻

**Review by phiz71 — APPROVED** [review]

**Pull request merged**